### PR TITLE
AsyncMessageProcessingConsumer countdown when the container is not active

### PR DESCRIPTION
**Pull request by renyansongno1**
Resolves https://github.com/spring-projects/spring-amqp/issues/1487

### CODE REVIEW

**Review by artembilan — APPROVED** [review]
LGTM, but is there a chance to add some unit test to confirm the problem and the fix?

Thanks

**Comment by renyansongno1** [comment]
@artembilan Hi, I have submitted the test code, please review.

File: spring-rabbit/src/test/java/org/springframework/amqp/rabbit/listener/SimpleMessageListenerContainerTests.java [hunk 1]
```diff
@@ -784,4 +809,33 @@ protected void doRollback(DefaultTransactionStatus status) throws TransactionExc
 
 	}
 
+	@SuppressWarnings("serial")
+	private class TestExecutor extends SimpleAsyncTaskExecutor {
```
Comment (artembilan) [inline spring-rabbit/src/test/java/org/springframework/amqp/rabbit/listener/SimpleMessageListenerContainerTests.java:813]:
```suggestion
	private static final class TestExecutor extends SimpleAsyncTaskExecutor {
```

Also, please, add your name to the `@author` list of affected classes.

**Review by artembilan — CHANGES_REQUESTED** [review]

**Comment by renyansongno1** [comment]
@artembilan Thank you for your time!

**Comment by artembilan** [comment]
... and cherry-picked to `2.4.x`

@renyansongno1 ,

thank you very much for the contribution; looking forward for more!

**Pull request merged**